src/test/java/ch/wisv/areafiftylan/EventRestIntegrationTest.java

```diff
@@ -0,0 +1,769 @@
+            body("object.eventName", is(eventDTO.get("name"))).
+            body("object.teamSize", equalTo(testEvent.getTeamSize())).
+            body("object.teamLimit", equalTo(testEvent.getTeamLimit()));
+        //@formatter:on
+
+        Event changedEvent = eventRepository.findOne(testEvent.getId());
+        Assert.assertEquals(eventDTO.get("name"), changedEvent.getEventName());
+    }
+
+    @Test
+    public void testChangeEventLimitOnly() {
+        Map<String, String> eventDTO = new HashMap<>();
+        eventDTO.put("teamLimit", "11");
+
+        SessionData login = login("admin");
+
+        //@formatter:off
+        given().
+            filter(sessionFilter).
+            header(login.getCsrfHeader()).
+        when().
+            content(eventDTO).contentType(ContentType.JSON).
+            put(EVENTS_ENDPOINT + "/" + testEvent.getId()).
+        then().
+            statusCode(HttpStatus.SC_OK).
+            body("object.eventName", is(testEvent.getEventName())).
+            body("object.teamSize", equalTo(testEvent.getTeamSize())).
+            body("object.teamLimit", equalTo(Integer.valueOf(eventDTO.get("teamLimit"))));
+        //@formatter:on
+
+        Event changedEvent = eventRepository.findOne(testEvent.getId());
+        Assert.assertEquals(eventDTO.get("teamLimit"), String.valueOf(changedEvent.getTeamLimit()));
+    }
+
+    @Test
+    public void testChangeEventTeamSizeOnly() {
+        Map<String, String> eventDTO = new HashMap<>();
+        eventDTO.put("teamSize", "3");
+
+        SessionData login = login("admin");
+
+        //@formatter:off
+        given().
+            filter(sessionFilter).
+            header(login.getCsrfHeader()).
+        when().
+            content(eventDTO).contentType(ContentType.JSON).
+            put(EVENTS_ENDPOINT + "/" + testEvent.getId()).
+        then().
+            statusCode(HttpStatus.SC_OK).
+            body("object.eventName", is(testEvent.getEventName())).
+            body("object.teamSize", equalTo(Integer.valueOf(eventDTO.get("teamSize")))).
+            body("object.teamLimit", equalTo(testEvent.getTeamLimit()));
+        //@formatter:on
+
+        Event changedEvent = eventRepository.findOne(testEvent.getId());
+        Assert.assertEquals(eventDTO.get("teamSize"), String.valueOf(changedEvent.getTeamSize()));
+    }
+
+    @Test
+    public void testChangeEventNonExistingId() {
+        Map<String, String> eventDTO = new HashMap<>();
+        eventDTO.put("name", "Changed Event");
+        eventDTO.put("teamSize", "3");
+        eventDTO.put("teamLimit", "11");
+
+        SessionData login = login("admin");
+
+        //@formatter:off
+        given().
+            filter(sessionFilter).
+            header(login.getCsrfHeader()).
+        when().
+            content(eventDTO).contentType(ContentType.JSON).
+            put(EVENTS_ENDPOINT + "/999").
+        then().
+            statusCode(HttpStatus.SC_BAD_REQUEST);
+        //@formatter:on
```

Comment:
Also weird case, but maybe check the original event to be sure the data hasn't actually changed. Returning a bad request is not a assurance that the internal hasn't actually changed.